src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -117,8 +117,17 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
 
         // cache the core classes
         if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+            if (true !== @mkdir($cache, 0777, true)) {
+                $error = error_get_last();
+                if (!is_dir($cache)) {
+                    if ($error) {
+                        throw new \RuntimeException(sprintf('Failed to create "%s": %s.', $cache, $error['message']));
+                    }
+                    throw new \RuntimeException(sprintf('Failed to create "%s".', $cache));
```

Comment:
Do you know if this case really exists? To me, when mkdir returns false, a warning is always triggered, which means error_get_last() always returns something. If you can't find a case where this happen, I'd suggest simplifying the code.